ond.py

```diff
@@ -90,27 +90,28 @@ def run_protocol(self, config_):
                         "FeatureExtraction", file_list
                     )
 
-                    results = {}
-                    results["detection"] = algorithm.execute(
-                        "WorldDetection", features_dict, logits_dict,
-                        red_light, round_id=round_id
-                    )
-                    results["classification"] = algorithm.execute(
-                        "NoveltyClassification", features_dict, logits_dict,
-                        round_id=round_id
-                    )
+                    if not config["feature_extraction_only"]:
+                        results = {}
+                        results["detection"] = algorithm.execute(
+                            "WorldDetection", features_dict, logits_dict,
+                            red_light, round_id=round_id
+                        )
+                        results["classification"] = algorithm.execute(
+                            "NoveltyClassification", features_dict, logits_dict,
+                            round_id=round_id
+                        )
 
-                    if config["use_feedback"]:
-                        algorithm.execute("NoveltyAdaption", None)
+                        if config["use_feedback"]:
+                            algorithm.execute("NoveltyAdaption", None)
 
-                    if config["save_features"]:
-                        logging.info(
-                            f"Writing features to {config['save_features']}"
-                        )
+                        if config["save_features"]:
```

Comment:
Shouldn't this be moved out of the feature extraction only we would still need to save the features if we are only doing feature extraction?

Reply:
Good point, I'll change this (or address it in another PR, since I think there are a few other places where saving features needs to be addressed).